mesop/examples/web_component/leaflet/leaflet_component.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Callable, Tuple, List, Dict
+
+import mesop.labs as mel
+
+
+@mel.web_component(path="./leaflet_map_component.js")
+def leaflet_map_component(
+    *,
+    center: Tuple[float, float] = (0.0, 0.0),  # [latitude, longitude]
+    zoom: int = 13,
+    markers: List[Dict[str, Any]] | None = None,  # List of marker objects
+    on_click: Callable[[mel.WebEvent], Any] | None = None,
+    key: str | None = None,
+):
+    """
+    A web component wrapper for a Leaflet map.
+
+    Args:
+        center: Initial map center as a tuple (latitude, longitude).
+        zoom: Initial map zoom level.
+        markers: A list of marker objects, each a dictionary with keys:
+            - latlng: Tuple (latitude, longitude) for marker position.
+            - popup_content: String content for the marker's popup (optional).
+        on_click: Callback function when the map is clicked (optional).
+        key: Unique key for the component.
+    """
+    events = {}
```

Comment:
Checking for None is no longer needed (See https://github.com/google/mesop/pull/1193). This is not in Google3 yet though.

So you should be able to do something like:

```
events={"clickEvent": on_click}
```

Reply:
Thanks for the suggestion. I made the changed and it worked well locally